js/ui/internetSearch.js

```diff
@@ -55,23 +55,42 @@ const InternetSearchProvider = new Lang.Class({
 
         try {
             parser.load_from_file(path);
+        } catch(e if e.matches(GLib.FileError, GLib.FileError.NOENT)) {
+            // User has not run Chromium yet.
+            return null;
         } catch (e) {
             logError(e, 'error while parsing Chromium preferences');
             return null;
         }
 
         let root = parser.get_root().get_object();
-        let searchProviderData = root.get_object_member('default_search_provider_data');
+
+        let searchProviderDataNode = root.get_member('default_search_provider_data');
+        if (!searchProviderDataNode || searchProviderDataNode.get_node_type() != Json.NodeType.OBJECT) {
+            return null;
+        }
+
+        let searchProviderData = searchProviderDataNode.get_object();
         if (!searchProviderData) {
             return null;
         }
 
-        let templateUrlData = searchProviderData.get_object_member('template_url_data');
+        let templateUrlDataNode = searchProviderData.get_member('template_url_data');
+        if (!templateUrlDataNode || templateUrlDataNode.get_node_type() != Json.NodeType.OBJECT) {
+            return null;
+        }
+
+        let templateUrlData = templateUrlDataNode.get_object();
         if (!templateUrlData) {
             return null;
         }
 
-        return templateUrlData.get_string_member('short_name');
+        let shortNameNode = templateUrlData.get_member('short_name');
+        if (!shortNameNode || shortNameNode.get_node_type() != Json.NodeType.VALUE) {
+            return null;
+        }
+
+        return shortNameNode.get_string();
```

Comment:
Is there any reason why we're using JSON-GLib here as opposed to just using JavaScript's native JSON.parse method? It seems like this results in overly verbose checks.